### PR TITLE
Issue #2747829 - Test order summary in checkout

### PR DESCRIPTION
**Pull request by ndf**
Work-in-progress

### CODE REVIEW

**Comment by ndf** [comment]
1. The order summary view can be turned on and off via checkout flow configuration
2. The order_summary_view view can be configured
3. Checkout summary is not shown on login/continue as guest, is shown on order information and payment

Are all in the tests.

@Lendude reviewed a little while working.

File: modules/checkout/tests/src/Functional/CheckoutOrderTest.php [hunk 1]
```diff
@@ -86,8 +97,39 @@ public function testGuestOrderCheckout() {
     ], 'Continue to review');
     $this->assertSession()->pageTextContains('Contact information');
     $this->assertSession()->pageTextContains('Billing information');
+    $this->assertSession()->pageTextContains('Order Summary');
```
Comment (mglaman) [inline modules/checkout/tests/src/Functional/CheckoutOrderTest.php:100]: We should test it is summarzing the orders contents (price, etc.)

**Comment by mglaman** [comment]
Also dependent on #423

**Comment by mglaman** [comment]
Closed for #449 which is merged and up to date.

**Pull request closed without merging**